File: Java/deepspace/src/deepspace/BetaPowerEfficientSpaceStation.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;

/**
 * Esta clase representa  el concepto de estación espacial eficiente beta así
 * como sus atributos y funcionalidades
 * @authors Manuel Diaz-Meco Terrés y Ana Graciani Donaire
 */
public class BetaPowerEfficientSpaceStation extends PowerEfficientSpaceStation {
    private final static float EXTRAEFFICIENCY = 1.2f;
    private Dice dado = null;
    
    BetaPowerEfficientSpaceStation(SpaceStation station) {
        super(station);
        dado = new Dice();
    }
        
    @Override
    public float fire() {
        if (dado.extraEfficiency())
            return super.fire()*EXTRAEFFICIENCY;
        else
            return super.fire();
    }
    
    @Override
    public BetaPowerEfficientSpaceStationToUI getToUI() {
        return new BetaPowerEfficientSpaceStationToUI(this);
    }
}

File: Ruby/BetaPowerEfficientSpaceStation.rb
#encoding: UTF-8

# Autores: Manuel Díaz-Meco Terrés y Ana Graciani Donaire
# 2ºDGIIM		PDOO

# Esta clase representa el concepto de estación espacial 
# eficiente beta así como sus atributos y funcionalidades

require_relative 'Dice'
require_relative 'PowerEfficientSpaceStation'
require_relative 'BetaPowerEfficientSpaceStationToUI'

module Deepspace
	class BetaPowerEfficientSpaceStation < PowerEfficientSpaceStation
		
		private
		
		@@EXTRAEFFICIENCY = 1.2
		
		public
		
		def initialize(spacestation)
			super(spacestation)
			@dice = Dice.new
		end
			
		def fire
			if @dice.extraEfficiency
				return super*@@EXTRAEFFICIENCY
			else
				return super
			end	
		end	
		
		def getUIVersion
			return BetaPowerEfficientSpaceStationToUI.new(this)
		end
		
	end
end
